pkg/test/kube/accessor.go

```diff
@@ -980,3 +988,67 @@ func (d *yamlDoc) toTempFile(workDir, fileNamePrefix string) (string, error) {
 	}
 	return name, nil
 }
+
+func newClientGetter(restConfig *rest.Config) resource.RESTClientGetter {
+	return &clientGetter{
+		restConfig: restConfig,
+	}
+}
+
+var _ resource.RESTClientGetter = &clientGetter{}
+
+type clientGetter struct {
+	restConfig *rest.Config
+}
+
+func (c *clientGetter) ToRESTConfig() (*rest.Config, error) {
+	return c.restConfig, nil
+}
+
+func (c *clientGetter) ToDiscoveryClient() (discovery.CachedDiscoveryInterface, error) {
+	return newDiscoveryClient(c.restConfig)
+}
+
+func (c *clientGetter) ToRESTMapper() (meta.RESTMapper, error) {
+	discoveryClient, err := c.ToDiscoveryClient()
+	if err != nil {
+		return nil, err
+	}
+	mapper := restmapper.NewDeferredDiscoveryRESTMapper(discoveryClient)
+	expander := restmapper.NewShortcutExpander(mapper, discoveryClient)
+	return expander, nil
+}
+
+func newDiscoveryClient(restConfig *rest.Config) (discovery.CachedDiscoveryInterface, error) {
+	// Copy the rest config.
+	cfg := *restConfig
+
+	// The more groups you have, the more discovery requests you need to make.
+	// given 25 groups (our groups + a few custom resources) with one-ish version each, discovery needs to make 50 requests
+	// double it just so we don't end up here again for a while.  This config is only used for discovery.
+	cfg.Burst = 100
+
+	parentDir := filepath.Join(homedir.HomeDir(), ".kube", "cache", "discovery")
```

Comment:
this is a bit suspicious, is there not a higher level part of the k8s library we can rely on?